source/agora/common/Transaction.d

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+
+    Defines the data structure of a transaction
+
+    The current design is heavily influenced by Bitcoin: as we have an UTXO,
+    starting with a simple input/output approach seems to make the most sense.
+    Script is not implemented: instead, we currently only do a simple signature
+    verification.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.Transaction;
+
+import agora.common.Data;
+import agora.common.crypto.Key;
+
+
+/*******************************************************************************
+
+    Currency amount to use
+
+    This is simply an alias to `long` at the moment, and should be made smarter.
+    Currently this has two major drawbacks:
+    - It does not do any overflow / underflow checks (over the total amount)
+    - It can go negative
+    We should probably wrap this in a smarter type to do currency operations,
+    and move it to a shared location, as currency operations might be performed
+    independently of `Transaction`.
+
+*******************************************************************************/
+
+public alias Amount = long;
+
+/*******************************************************************************
+
+    Represents a transaction (shortened as 'tx')
+
+    Agora uses an UTXO model for transactions, so the transaction format is
+    originally derived from that of Bitcoin.
+
+*******************************************************************************/
+
+public struct Transaction
+{
+    /// The list of `outputs` from previous transaction(s) being spent
+    Input[] inputs;
+    /// The list of newly created outputs to put in the UTXO
+    Output[] outputs;
+}
+
+/*******************************************************************************
+
+    Represents an entry in the UTXO
+
+    This is created by a valid `Transaction` and is added to the UTXO after
+    a transaction is confirmed.
```

Comment:
What does it mean that a transaction is _confirmed_, in this context? It should be documented.

Reply:
I think that's pretty standard Bitcoin jargon.

Reply:
I still don't understand though, what does it mean in this context? I know about the "6 confirmations rule", but is that related to this at all?

Reply:
Confirmed == Included in a block

Reply:
got it!